machine-learning/01-openpbta_heatmap.Rmd

```diff
@@ -0,0 +1,383 @@
+
+```{r random_z_sd, live = TRUE}
+sd(zscores_mat[25, ])
+```
+
+The standard deviation is now 1.
+That means all the variance values should be equal to one, too.
+
+```{r summary_row_variance}
+summary(matrixStats::rowVars(zscores_mat))
+```
+
+If you're filtering to high variance values, it's important to do that _prior_ to standardizing.
+
+### Heatmap itself!
+
+Okay, now we're ready to make a heatmap complete with annotation bars.
+
+```{r make_heatmap}
+Heatmap(zscores_mat,
+        # The distance metric used for clustering the rows
+        clustering_distance_rows = "pearson",
+        # Linkage method for row clustering
+        clustering_method_rows = "average",
+        # Distance metric for columns
+        clustering_distance_columns = "pearson",
+        # Linkage for columns
+        clustering_method_columns = "average",
+        show_row_names = FALSE,
+        show_column_names = FALSE,
+        # Add annotation bars to the top of the 
+        top_annotation = column_annotation,
+        # This will be used as the label for the color bar
+        # of the cells of the heatmap itself
+        name = "z-score")
+```
+
+```{r save_heatmap_png, live = TRUE}
+png(filename = heatmap_file, 
+    width = 11, 
+    height = 7, 
+    units = "in", 
+    res = 300)
+Heatmap(zscores_mat,
+        clustering_distance_rows = "pearson",
+        clustering_method_rows = "average",
+        clustering_distance_columns = "pearson",
+        clustering_method_columns = "average",
+        show_row_names = FALSE,
+        show_column_names = FALSE,
+        top_annotation = column_annotation,
+        name = "z-score")
+dev.off()
+```
+
+## Spotlight on medulloblastoma
+
+The heatmap we made with all of the histologies under consideration suggested that there's some structure _within_ histologies.
+Let's take a closer look at the medulloblastoma samples.
+
+```{r mb_sample_ids, live = TRUE}
+medulloblastoma_samples <- filtered_histologies_df %>%
+  filter(short_histology == "Medulloblastoma") %>%
+  pull(Kids_First_Biospecimen_ID)
+```
+
+Use all genes to create a matrix that only contains medulloblastoma samples.
+
+```{r mb_mat, live = TRUE}
+medulloblastoma_mat <- rnaseq_mat[, medulloblastoma_samples]
+```
+
+We'll use Pearson correlation for our distance metric here again, but let's breakdown what goes into hierarchical clustering a bit here.
+
+```{r calculate_cor}
+# cor() will take the correlation between columns - we're interested in the 
+# relationship between medulloblastoma samples, so we don't need to transpose
+# the matrix before using cor()
+mb_sample_correlation <- cor(medulloblastoma_mat,
+                             use = "pairwise.complete.obs",
+                             method = "pearson")
+```
+
+Hierarchical clustering is performed on dissimilarities, so we'll subtract the correlation values from 1.
+To use `hclust()` -- the built-in method for hierarchical clustering in R -- we need a dissimilarity structure.
```

Comment:
I feel like these sentences need a bit more clarity. What we need is a distance matrix... how is this different from just the (1-cor) matrix?

Reply:
I'll talk about what `as.dist()` does explicitly.